src/main/java/org/mitre/synthea/engine/Logic.java

```diff
@@ -191,8 +193,16 @@ public boolean test(Person person, long time) {
           return false;
         }
       }
-
-      return Utilities.compare(observation.value, this.value, operator);
+      if (valueCode != null) {
+        value = valueCode;
+      } 
+      if (operator == "is nil") {
```

Comment:
In Java, String equality testing should be done with `str1.equals(str2)` , the exception being when you want to check for null. The short version is that == checks whether 2 variables point to the same object in memory, `.equals` checks whether 2 variables point to objects which have the same contents. This is a good overview:  https://stackoverflow.com/a/513839 They actually say use `Objects.equals` which is ok too but not as common.

So these should be `operator.equals("is nil")` and `operator.equals("is not nil")`